src/main/java/de/siegmar/logbackgelf/GelfEncoder.java

```diff
@@ -216,8 +232,22 @@ private void addField(final Map<String, Object> dst, final String key, final Str
             addWarn("staticField key '" + key + "' is illegal. "
                 + "Keys must apply to regex ^[\\w.-]*$");
         } else {
-            dst.put(key, value);
+            final Object processedValue = processValue(key, value);
+            if (processedValue != null) {
+                dst.put(key, processedValue);
+            }
+        }
+    }
+
+    private Object processValue(final String key, final String value) {
+        if (integerFields.contains(key)) {
+            if (value.matches("\\d+")) {
```

Comment:
I'd feel better if this is not done via (slow) regexp but via a simple `isNumber()` method that just checks content for 0-9.

Reply:
`isNumber()` is not a method of `String`

Reply:
I meant to implement something. But at least the regexp should be `-?\\d+`.